Userland/Libraries/LibGfx/ImageFormats/BMPLoader.cpp

```diff
@@ -1372,7 +1372,9 @@ static ErrorOr<void> decode_bmp_pixel_data(BMPLoadingContext& context)
             case 24: {
                 if (!streamer.has_u24())
                     return Error::from_string_literal("Cannot read 24 bits");
-                context.bitmap->scanline(row)[column++] = streamer.read_u24();
+                auto color = streamer.read_u24();
```

Comment:
This change doesn't make a lot of sense on its own IMO. When the bitmap is created, it is explicitly mentioned that the type is BGR (without alpha). 

From here there are two ways IMO:
 - We ditch BGR and always use BGRA. (So basically the route you started to take).
 - We use RGB instead of BGR, which would also make a lot of sense as this is what people use.

Reply:
I see it this way: The BMP has no alpha channel but the alpha byte per pixel is reserved anyway. This is represented in the format BGRx8888. Setting the alpha byte to opaque has no disadvantage, as before it was uninitialized. The advantage is, that it allows us to "convert" the bitmap to BGRA8888 with zero costs. Loading the BMP as BGRA8888 would be not correct in my opinion, because then the user of the bitmap is required to handle the fully opaque alpha channel.

Reply:
I agree with @simonkrauter here; the bitmap format carries some information on the intent of the decoded bitmap, such as in this case, whether it has an alpha channel or not. Always decoding as BGRA removes that intent.

Not sure how RGB vs. BGR is going to help with alpha handling?

BGRx handling goes well if either the producer or consumer of the bitmap uses the existing pixel color APIs. Since that is usually slow, we get a pointer to the scanline and deal with the raw bitmap data ourselves. It feels wrong to either initialize or assume a channel that is basically non-existent, but it does automatically make all consumers do the right thing so :shrug:

Reply:
> I agree with @simonkrauter here; the bitmap format carries some information on the intent of the decoded bitmap, such as in this case, whether it has an alpha channel or not. Always decoding as BGRA removes that intent.

Note that we pass it as BGRA to Skia, so you lose the intent as well, just later on.

> Not sure how RGB vs. BGR is going to help with alpha handling? Or do you mean we get to use kRGBA_8888_SkColorType vs kRGB_888x_SkColorType?

Yes, if there is no alpha channel on the image, we shouldn't pass an alpha-aware color type to Skia. Which reduce our choice to only `kRGB_888x_SkColorType`.

---
Right now, we have multiple producer and one consumer for bitmaps. With this change, we will have one bitmap producer that produce a bitmap variant that can be converted to BGRA for free and some other that don't. So, I guess that soon, you will need to either convert all producers to do this behavior or have a way to distinguish between fast-to-convert Bitmaps and not-fast-to-convert Bitmaps. I really don't like that.

Another solution, which I think is preferable, is to stop generating Bitmaps in a format that none of our consumer use (BGR) and replace it with RGB. This way, there is no need to even think about smuggling an alpha channel into BGR bitmaps.

Reply:
> Note that we pass it as BGRA to Skia, so you lose the intent as well, just later on.

Absolutely, at the moment it's more of a "keep the intent alive for as long as possible" exercise than anything else.

I agree with you that using RGB(A) as an output type makes more sense for Ladybird since Skia (and possibly other rendering libraries as well) natively supports both the version with and without the alpha channel. On top of that, I don't believe there are performance issues with using RGBA vs. BGRA nowadays, which was the main reason to use BGRA - it's supposed to be in the same byte order as the graphics adapter uses internally.

Reply:
Could you please help me with https://github.com/LadybirdBrowser/ladybird/pull/2132?